### PR TITLE
API Payload Compression

### PR DESCRIPTION
**Pull request by bcongdon**
## Description
 
* Addresses #1360 -- adds ability to enable payload compression on Lambda Rest APIs
* New setting `payload_compression` for enabling/disabling payload compression
* New setting `payload_minimum_compression_size` to set the threshold for compressing the payload

Payload compression is a property of the REST api -- like binary support -- so it has to be enabled/disabled independently from API stages

## GitHub Issues

#1360

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -830,6 +830,8 @@ to change Zappa's behavior. Use these at your own risk!
         "log_level": "DEBUG", // Set the Zappa log level. Can be one of CRITICAL, ERROR, WARNING, INFO and DEBUG. Default: DEBUG
         "manage_roles": true, // Have Zappa automatically create and define IAM execution roles and policies. Default true. If false, you must define your own IAM Role and role_name setting.
         "memory_size": 512, // Lambda function memory in MB. Default 512.
+        "payload_compression": False, // Whether or not to enable API gateway payload compression (default: False)
```
Comment (Miserlou) [inline README.md]: This should be `false`, not `False`.

**Comment by Miserlou** [comment]
Nice! Small doc change please. 

Is there any reason not to enable this by default?

**Comment by bcongdon** [comment]
Hmm... I guess it does seem sensible to enable by default.

It [looks like](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-receive-response-with-compressed-payload.html) the compression is only done if an appropriate `Accept-Encoding` header is set, so it should be safe.

**Comment by rgov** [comment]
See my comment in #1360:

> I think the documentation should reference security trade-offs with regard to the [CRIME attack](https://en.wikipedia.org/wiki/CRIME).

I am not sure it should be turned on by default in light of this attack.

**Pull request merged**